coral/src/app/components/ComplexNativeSelect.tsx

```diff
@@ -0,0 +1,88 @@
+import { NativeSelect, NativeSelectProps } from "@aivenio/aquarium";
+import { omit } from "lodash";
+import { ChangeEvent, useState } from "react";
+
+type ComplexNativeSelectProps<ComplexNativeSelectOptionType> =
+  NativeSelectProps & {
+    options: Array<ComplexNativeSelectOptionType>;
+    identifierValue: keyof ComplexNativeSelectOptionType;
```

Comment:
For the sake of consistency, we should use the same prop type (`getValue?: (opt: T) => string;`) to parse option value as in [Select.tsx](https://github.com/aiven/design-system/blob/main/src/components/Select/Select.tsx). It is possible that we might have objects with a composite key in DB and using that as the value would be safest option.

Reply:
I thought it's nice if more responsibility was moved to the component itself instead of being handled outside, since we have a specific use case we want to cover. I normally try to not make things complex from the beginning to cover possible future use cases (that may never occur). But in our case it also makes sense to match more closely to existing implementations like the DS.

Reply:
> Using a function is less typesafe

As the function uses a generic type, I am not able to see how it is not typesafe.

> and making it more reusable if we really need it to be.

Already in the current AngularJS UI we have the requirement. We have select where one of the options represents the cluster default and that value is hilighted with `(default)` suffix in the option label

Normally I am a big believer of YAGNI, but in this case, I would opt for complexity from day 1. Lets say way keep the current `pick` approach and we later identify the need for computed values. What is our migration strategy? Would we add a new prop called  `getValue()` that takes precedence over `identifierValue `? Would they be mutually exclusive or would be simply change the type of `identifierValue` from string into function and fix all occurrences then?

Reply:
> As the function uses a generic type, I am not able to see how it is not typesafe.
Ah, that was written poorly by me. I meant less safe in a way that the component itself can't make sure that the value returned from the function is a valid value for the option. 

>What is our migration strategy? Would we add a new prop called getValue() that takes precedence over identifierValue ? Would they be mutually exclusive or would be simply change the type of identifierValue from string into function and fix all occurrences then?
Since we're not a library, we don't really have the need to make sure prop like `identifierValue` is working. I would remove `identifierValue` (and name), add the new prop and behaviour and replace it in existing usage!

>Already in the current AngularJS UI we have the requirement. We have select where one of the options represents the cluster default and that value is hilighted with (default) suffix in the option label
If we already know we have the requirement, I can update the component of course!

Would it be better for you we merge the current state so you can finish your ticket and make a follow up PR or should we rather update it right now? Don't want to block you!

Reply:
Sounds good. 🚢 - it.

Reply:
Added a follow up ticket! https://github.com/aiven/klaw/issues/410